File: Gosuji/Gosuji.Client/Models/Trainer/Score.cs
namespace Gosuji.Client.Models.Trainer
{
    public class Score
    {
        public int Winrate { get; set; }
        public int ScoreLead { get; set; }

        public Score() { }

        public Score(int winrate, int scoreLead)
        {
            Winrate = winrate;
            ScoreLead = scoreLead;
        }

        public string FormatWinrate(bool shouldReverse = false)
        {
            return ((shouldReverse ? GetReverseWinrate() : Winrate) / 1_000_000.0).ToString("F2");
        }

        public string FormatScoreLead(bool shouldReverse = false)
        {
            return ((shouldReverse ? GetReverseScoreLead() : ScoreLead) / 1_000_000.0).ToString("F1");
        }

        public int GetReverseWinrate()
        {
            return 100_000_000 - Winrate;
        }

        public int GetReverseScoreLead()
        {
            return ScoreLead * -1;
        }
    }
}

File: Gosuji/Gosuji.API/Resources/js/pages/trainer/classes/Score.js
class Score {
    winrate;
    scoreLead;


    constructor(winrate, scoreLead) {
        this.winrate = winrate;
        this.scoreLead = scoreLead;
    }


    formatWinrate(shouldReverse = false) {
        return (shouldReverse ? this.getReverseWinrate() : this.winrate).toFixed(2);
    }

    formatScoreLead(shouldReverse = false) {
        return (shouldReverse ? this.getReverseScoreLead() : this.scoreLead).toFixed(1);
    }

    reverse() {
        this.winrate = this.getReverseWinrate();
        this.scoreLead = this.getReverseScoreLead();
        return this;
    }

    getReverseWinrate() {
        return 100 - this.winrate;
    }

    getReverseScoreLead() {
        return this.scoreLead * -1;
    }

    copy() {
        return new Score(this.winrate, this.scoreLead);
    }
}

if (!window.trainer) window.trainer = {};
if (g.DEBUG && !window.trainer.Score) window.trainer.Score = Score;

export { Score };
